vision/cloud-client/src/main/java/com/example/vision/Detect.java

```diff
@@ -0,0 +1,405 @@
+        detectProperties(path, System.out);
+      }
+    } else if (command.equals("safe-search")) {
+      if (path.startsWith("gs://")) {
+        // TODO: See https://goo.gl/uWgYhQ
+      } else {
+        detectSafeSearch(path, System.out);
+      }
+    }
+  }
+
+  private static ImageAnnotatorClient visionApi;
+
+  /**
+   * Constructs a {@link Detect} which connects to the Cloud Vision API.
+   */
+  public Detect(ImageAnnotatorClient client) {
+    visionApi = client;
+  }
+
+  /**
+   * Helper for getting byte array given input stream.
+   * @param is Input stream to get bytes for.
+   * @return Byte array for the input stream contents.
+   * @throws IOException
+   */
+  static public byte[] toByteArray(InputStream is) throws IOException {
+    ByteArrayOutputStream buffer = new ByteArrayOutputStream();
+
+    int nRead;
+    byte[] data = new byte[16384];
+
+    while ((nRead = is.read(data, 0, data.length)) != -1) {
+      buffer.write(data, 0, nRead);
+    }
+
+    buffer.flush();
+
+    return buffer.toByteArray();
+  }
+  
+  /**
+   * Detects faces in the specified image.
+   * @param filePath The path to the file to perform face detection on.
+   * @param out A {@link PrintStream} to write detected features to.
+   * @throws IOException 
+   */
+  static public void detectFaces(String filePath, PrintStream out) throws IOException {
+    List<AnnotateImageRequest> requests = new ArrayList<>();
+
+    FileInputStream file = new FileInputStream(filePath);
+    ByteString imgBytes = ByteString.copyFrom(toByteArray(file));
+    
+    Image img = Image.newBuilder().setContent(imgBytes).build();    
+    Feature feat = Feature.newBuilder().setType(Type.FACE_DETECTION).build();
+    AnnotateImageRequest request = AnnotateImageRequest.newBuilder()
+        .addFeatures(feat)
+        .setImage(img)
+        .build();
+    requests.add(request);
+
+    BatchAnnotateImagesResponse response = visionApi.batchAnnotateImages(requests);
+    List<AnnotateImageResponse> responses = response.getResponsesList();
+
+    for (AnnotateImageResponse res : responses) {      
+      if (res.hasError()) out.printf("Error: %s\n", res.getError().getMessage());
+
+      for (FaceAnnotation annotation : res.getFaceAnnotationsList()) {
+        out.printf("anger: %s\njoy: %s\nsurprise: %s\n",
+            annotation.getAngerLikelihood(),
+            annotation.getJoyLikelihood(),
+            annotation.getSurpriseLikelihood());
+      }
+    }
+  }
+  
+  /**
+   * Detects labels in the specified image.
+   * @param filePath The path to the file to perform label detection on.
+   * @param out A {@link PrintStream} to write detected labels to.
+   * @throws IOException 
+   */
+  static public void detectLabels(String filePath, PrintStream out) throws IOException {
+    List<AnnotateImageRequest> requests = new ArrayList<>();
+
+    FileInputStream file = new FileInputStream(filePath);
+    ByteString imgBytes = ByteString.copyFrom(toByteArray(file));
```

Comment:
Couldn't you use [`ByteString.readFrom(file)`](https://developers.google.com/protocol-buffers/docs/reference/java/com/google/protobuf/ByteString#readFrom-java.io.InputStream) directly?